### ISSUE TITLE
Cannot use set_local with a Vector of a mixed function

### ISSUE THREAD

**Issue by ctjacobs** [firedrakeproject/firedrake#460]
For some given array (of arrays, in the case of the MFS here), I'm trying to assign it to the vector of `f` using `set_local`:

```
from firedrake import *

mesh = UnitSquareMesh(1, 1)
U = FunctionSpace(mesh, "CG", 2)
H = FunctionSpace(mesh, "CG", 1)
W = MixedFunctionSpace([U, H])

f = Function(W)
f.interpolate(Expression(("0.1", "0.1")))

g = Function(W)
g.interpolate(Expression(("2.0", "1.0")))

f.vector().set_local(g.vector().array())
```

but this fails with:

```
Traceback (most recent call last):
  File "test_mfs_vector_assign.py", line 14, in <module>
    f.vector().set_local(g.vector().array())
  File "/data/firedrake/firedrake/vector.py", line 52, in set_local
    self.dat.data[:] = values
TypeError: 'tuple' object does not support item assignment
```

**Comment by wence-** [comment]
Are you always going to be setting from an array() obtained from a vector()?

**Comment by wence-** [comment]
In general I'm slightly concerned about the usage of `.vector()` and mixed functions.  In various places dolfin_adjoint looks like it can just assign to a slice of a `.vector()` but that's not currently possible for mixed functions.

**Comment by danshapero** [comment]
This was fixed in #489.
